src/context-providers/yup/types.ts

```diff
@@ -78,6 +78,11 @@ export interface IYupConditionalValidationRule extends IYupRule {
 export interface IYupRenderRule extends IYupRule {
 	filled?: boolean | undefined;
 	shown?: boolean | undefined;
+	withinDays?: {
+		numberOfDays: number;
+		fromDate?: string | undefined;
+		dateFormat?: string;
+	};
```

Comment:
create a new interface and apply it to `IYupRenderRule` and `IDateFieldValidationRule` and `calculateWithinDaysRange()`

```
export interface IWithinDaysRule {
	numberOfDays: number;
	fromDate?: string | undefined;
	dateFormat?: string | undefined;
}
```